test-app/test/JavaControllerTest.java

```diff
@@ -92,7 +86,7 @@ private TestUsernamePasswordAuthProvider upAuthProvider() {
 				.plugin(TestUsernamePasswordAuthProvider.class);
 	}
 
-	private UserServicePlugin userServicePlugin() {
+	private AbstractUserService userServicePlugin() {
 		return Play.application().plugin(TestUserServicePlugin.class);
```

Comment:
I managed to tackle that problem elegantly actually (not that I am that smart - simply everything was already there); basically I had `FakeApplication` so I could easily use it as:
`return app.injector().instanceOf(TestUserService.class);`

No need to call `Play.application`...